src/screens/login/Login.tsx

```diff
@@ -0,0 +1,125 @@
+import { useState } from 'react';
+import { Image, Text, View } from 'react-native';
+
+import Button from '@components/button';
+import Input from '@components/input';
+import { passwordRegex, userNameRegex } from '@constants/regex';
+import { storageKeys } from '@constants/storageKeys';
+import { useUserLoginContext } from '@contexts/LoginContext';
+import { logoImg, validateInput } from '@helpers/helper';
+import { addData, fetchData } from '@network/apiMethods';
+import { newSession, newToken, validateLogin } from '@network/apiUrls';
+import { storeData } from '@storage/storage';
+
+import { styles } from './styles';
+
+const LoginScreen = () => {
+  const { setLoginId } = useUserLoginContext();
+  const [inputs, setInputs] = useState({
+    userName: {
+      value: '',
+      isValid: true,
+      errorMessage: '',
+    },
+    password: {
+      value: '',
+      isValid: true,
+      errorMessage: '',
+    },
+  });
+
+  const handleInputChange = (inputIdentifier: string, enteredValue: string) => {
+    setInputs(currentInputs => {
+      return {
+        ...currentInputs,
+        [inputIdentifier]: { value: enteredValue, isValid: true },
+      };
+    });
+  };
+
+  const handleSubmit = async () => {
+    try {
+      const loginData = {
+        userName: inputs.userName.value,
+        password: inputs.password.value,
+      };
+
+      const userNameError = validateInput(
+        loginData.userName,
+        userNameRegex,
+        'userName is required.',
+        'Please enter a valid userName.',
+      );
+
+      const passwordError = validateInput(
+        loginData.password,
+        passwordRegex,
+        'Password is required.',
+        "Entered password doesn't match the password format.",
+      );
+
+      if (userNameError || passwordError) {
+        setInputs(currentInputs => ({
+          userName: {
+            ...currentInputs.userName,
+            isValid: !userNameError,
+            errorMessage: userNameError,
+          },
+          password: {
+            ...currentInputs.password,
+            isValid: !passwordError,
+            errorMessage: passwordError,
+          },
+        }));
+
+        return;
+      }
+
+      const reqToken = await fetchData(newToken);
+      const request_token = reqToken?.request_token;
+      const validLogin = await addData(validateLogin, {
+        request_token: request_token,
+        username: loginData.userName,
```

Comment:
at some places using "userName" and some places "username" why?

Reply:
because userName is a variable defined by me while username is the query param required by api.

Reply:
you can maintain same naming convention to avoid issues later.
Like when you want to pass username using destructuring
directly use like this-

const validLogin = await addData(validateLogin, {
        request_token: requestToken,
        username,
})